client/core/core.go

```diff
@@ -1670,7 +1693,7 @@ func (c *Core) SetWalletPassword(appPW []byte, assetID uint32, newPW []byte) err
 		return newError(authErr, "SetWalletPassword password error: %v", err)
 	}
 
-	newPasswordSet := len(newPW) > 0
+	newPasswordSet := len(newPW) > 0 // excludes empty but non-nil
```

Comment:
One thing that's a little confusing is that you're using the same name for this variable in both `SetWalletPassword` and `ReconfigureWallet`, but they have different meanings. Maybe try to capture their function more clearly in some new names, e.g. `isSettingNewPW` and `usingNonNilPW` or whatever.

Reply:
When I did the rebase for @amassarwi, i added these two comments to point out just that distinction.  I agree changing the names would be best.

Reply:
Agreed, will do

Reply:
Actually looks like `SetWalletPassword` shouldn't exclude empty strings, instead it should check if newPW
isn't nil as we do in `ReconfigureWallet`.

We check for non empty string inside `setWalletPassword`.